lxd/device/device_utils_instance.go

```diff
@@ -1,8 +1,18 @@
 package device
 
 import (
+	"github.com/lxc/lxd/lxd/project"
 	"github.com/lxc/lxd/lxd/state"
+	"github.com/lxc/lxd/shared"
 )
 
 // InstanceLoadNodeAll returns all local instance configs.
 var InstanceLoadNodeAll func(s *state.State) ([]InstanceIdentifier, error)
+
+// InstanceLoadByProjectAndName returns instance config by project and name.
+var InstanceLoadByProjectAndName func(s *state.State, project, name string) (InstanceIdentifier, error)
+
+// DevicesPath returns the path where devices are created for an instance.
```

Comment:
@stgraber this appears to be a container specific function, so have not added it to the InstanceIdentifier interface, also its only defined in containerLXC and not container. Unlike LogPath().

Reply:
I think we'll find ourselves using that for things like disk even with other types, so I'm fine with exposing it if that helps.

Reply:
@stgraber ok great, will add to interface.